api/src/wfl/server.clj

```diff
@@ -71,25 +71,33 @@
       wrap-internal-error
       (wrap-json-response {:pretty true})))
 
-(defn stfu-jetty
-  "Set up a stub logger to shut Jetty up."
-  []
-  (Log/setLog
-   (proxy [Logger] []
-     (debug ([thrown]) ([msg & args]))
-     (getLogger [name] this)
-     (getName [] "JettyStfu")
-     (ignore [ignored])
-     (info ([thrown]) ([msg & args]))
-     (isDebugEnabled [] false)
-     (setDebugEnabled [enabled])
-     (warn ([thrown]) ([msg & args])))))
+(defn ^:private start-workload-manager []
+  (letfn [(update-workloads! [])] ;; todo
+    (log/info "starting workload update loop")
+    (update-workloads!)
+    (future
+      (loop []
+        (update-workloads!)
+        (.sleep TimeUnit/SECONDS 20)
+        (recur)))))
+
+(defn ^:private start-server [args]
+  (let [port {:port (util/is-non-negative! (first args))}]
+    (log/info wfl/the-name port)
+    (future (jetty/run-jetty app port))))
+
+(defn ^:private await-some [& futures]
+  (loop []
+    (let [fs (filter future-done? futures)]
+      (if (not-empty fs)
+        @(first fs)
+        (do
+          (.sleep TimeUnit/SECONDS 1)
+          (recur))))))
```

Comment:
await-some is called once and returns when any future is done?
Do you intend for this to be like select(2)?

Reply:
I don't understand the question - i could have used select instead of filter I suppose.
The point is that I want to join both futures such that is one returns, the server closes.
Neither of these futures are meant to return - only if an error occurs then will they.